client/scala/src/main/scala/ai/verta/blobs/dataset/PathBlob.scala

```diff
@@ -0,0 +1,122 @@
+package ai.verta.blobs.dataset
+
+import ai.verta.swagger._public.modeldb.versioning.model._
+
+import java.security.{MessageDigest, DigestInputStream}
+import java.io.{File, FileInputStream}
+
+import scala.collection.mutable.HashMap
+import scala.util.{Failure, Success, Try}
+
+/** Captures metadata about files
+ *  @param paths list of filepaths or directory paths
+ */
+ case class PathBlob(private val paths: List[String]) extends Dataset {
+  private val BufferSize = 8192
+  paths.map(expanduser _).map((path: String) => processPath(new File(path)))
+
+  override def equals(other: Any) = other match {
+    case other: PathBlob => contents.equals(other.contents)
+    case _ => false
+  }
+
+  /** Hash the file's content
+   *  From https://stackoverflow.com/questions/41642595/scala-file-hashing
+   *  @param path filepath
+   */
+  private def hash(file: File): String = {
```

Comment:
Oh I was wrapping the Try down in the processFile or something, since there could be multiple method calls there that cause an exception (not just hash). But I'll add a Try here since in the future someone might re-use this, and should probably know it might cause an exception. Resolved.